hil/api.py

```diff
@@ -162,7 +163,7 @@ def network_revoke_project_access(project, network):
     """Remove access to <network> from <project>.
 
     If the project or network does not exist, a NotFoundError will be raised.
-    If the project is the owner of the network a BlockedError will be raised.
+    If the project is the owner of the network a AttachedResourceError will be raised.
```

Comment:
This seems conceptually wrong to me; this issue isn't that something needs detaching, but that something fundamentally *cannot* be detached.

Reply:
Yes exactly that's why I said some of the errors didn't quite fit into it so I named them ```AttachedResourceError``` for now. I think I wrote ```AuthorizationError``` previously for this but then had to redo all changes for some reason so I might've put the former. What do you think about ```AuthorizationError``` though?

Reply:
`AuthorizationError` isn't right either; it isn't a permissions issue. I think `BadArgumentError` is the closest thing we currently have. I'm not terribly opposed to adding a new type for this sort of thing.

@naved001, @ianballou, interested in your thoughts.

Reply:
I agree with @zenhack, `BadArgumentError` is the closest thing we have. This is the only case we have right now where we have the concept of ACLs. So I think `BadArgumentError` with a message explaining why the arguments are bad (cannot revoke access from project since project is the owner) is good enough.
If we were to create a new type we could call it `IllegalOperationError` since it's not legal to for a project to not have access to the network it created. And I think it would be generic enough to be used in similar situations that we may run into.

Reply:
Yeah I agree that `AttachedResourceError` doesn't work here. That error should alert the user that they need to detach something first before continuing with the command that raised the error.  Here, there's nothing that needs detaching.

I agree with @naved001 that something like `IllegalOperationError` would be good.  That plus a good error string stating why it's illegal.

Reply:
Either of those solutions is fine with me.

Reply:
Okay I'm a little confused here. We already have ```IllegalStateError``` to indicate the "generic" illegal operation. Is it serving any other purpose too? Also, according to the description, it is essentially doing the same thing as ```AttachedResourceError``` unless I'm missing out on something. Should we rename it to ```IllegalOperationError``` and give it a proper definition?

Reply:
`IllegalStateError` isn't quite just any generic illegal op. Right now I think it's only used with headnodes, in cases where somebody does something like try to modify the configuration of the VM while it's running -- it is in an "illegal state." The user could get things working by shutting the VM off first. Whereas with this case, removing the owner of a network is simply not allowed; there's nothing the user can do to make the operation legal.